cluster/addons/metadata-agent/stackdriver/metadata-agent-rbac.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: system:metadata-agent
```

Comment:
s/metadata-agent/stackdriver-metadata-agent/?

I can imagine more than one thing called metadata agent running in the cluster, so it may be better to explicitly tell which one we are referring to here. Same for service account name.

Reply:
There is no other system component called metadata agent right now, so it shouldn't collide. I prefer to keep it consistent with addon name and with other addons (e.g. you have cluster role "system:heapster", not "system:stackdriver-heapster").

Reply:
The DaemonSet name right now is stackdriver-agents, not metadata-agent. Maintaining consistency with container name doesn't seem to be a reasonable approach, since the same container name can be the same for different pods, while ServiceAccount name should be unique within a namespace.

Reply:
Changed everything to metadata-agent.

Reply:
But why? Since now you're changing the names everywhere, why not be explicit?

Reply:
My intention was to name the addon "metadata-agent", this is also reflected e.g. in directory layout and env variables. Here I just fixed some unintentional inconsistency.

Reply:
Ack.

Reply:
`system:` is reserved for roles setup by the API server. Please consider re-naming this to `stackdriver-metadata-agent`.

Reply:
What about `stackdriver:metadata-agent`? Would it make sense to settle on a `stackdriver` prefix (i.e., do we anticipate any more Stackdriver-related roles)?

Reply:
Either `stackdriver:metadata-agent` or `stackdriver-metadata-agent` seem fine.

Reply:
Sorry for delay. Introducing this fix now in #61394 - PTAL